common/djangoapps/student/models.py

```diff
@@ -333,7 +333,8 @@ def email_exists_or_retired(email):
     """
     Check an email against the User model for existence.
     """
-    return User.objects.filter(email=email).exists() or is_email_retired(email)
+    return User.objects.filter(email=email).exists() or is_email_retired(email) or \
+           AccountRecovery.objects.filter(secondary_email=email)
```

Comment:
We need to call the `.exists()` as well on it. Also, can make it a little cleaner?
```suggestion
    return (
        User.objects.filter(email=email).exists() or
        is_email_retired(email) or
        AccountRecovery.objects.filter(secondary_email=email).exists()
    )
```